switchlink/switchlink_link_test.cc

```diff
@@ -36,79 +36,96 @@ extern "C" {
-  switchlink_db_tunnel_interface_info_t tunnel_info;
-  switchlink_db_interface_info_t interface_info;
-  uint32_t ifindex;
+  union {
+    // Parameter values
+    switchlink_db_tunnel_interface_info_t tunnel_info;
+    switchlink_db_interface_info_t interface_info;
+    uint32_t ifindex;
+  };
   // Handler tracking
   enum handler_type handler;
-  int num_handler_calls;
 };
 
-static struct test_results results;
-
-
 /**
- * Dummy functions.
+ * Test results vector. Contains one entry for each dummy function called
+ * by the UUT.
  */
-void switchlink_create_interface(switchlink_db_interface_info_t *intf) {
-  results.handler = CREATE_INTERFACE;
-  results.num_handler_calls += 1;
+std::vector<test_results> results(2);
+
+//----------------------------------------------------------------------
+// Test doubles (dummy functions)
+//----------------------------------------------------------------------
+
+void switchlink_create_interface(switchlink_db_interface_info_t* intf) {
+  struct test_results temp = {0};
+  temp.handler = CREATE_INTERFACE;
   if (intf) {
-    results.interface_info = *intf;
+    temp.interface_info = *intf;
   }
+  results.push_back(temp);
 }
 
 void switchlink_delete_interface(uint32_t ifindex) {
-  results.handler = DELETE_INTERFACE;
-  results.num_handler_calls += 1;
-  results.ifindex = ifindex;
+  struct test_results temp = {0};
+  temp.handler = DELETE_INTERFACE;
+  temp.ifindex = ifindex;
+  results.push_back(temp);
 }
 
 void switchlink_create_tunnel_interface(
-    switchlink_db_tunnel_interface_info_t *tnl_intf) {
-  results.handler = CREATE_TUNNEL_INTERFACE;
-  results.num_handler_calls += 1;
+    switchlink_db_tunnel_interface_info_t* tnl_intf) {
+  struct test_results temp = {0};
+  temp.handler = CREATE_TUNNEL_INTERFACE;
   if (tnl_intf) {
-    results.tunnel_info = *tnl_intf;
+    temp.tunnel_info = *tnl_intf;
   }
+  results.push_back(temp);
 }
 
 void switchlink_delete_tunnel_interface(uint32_t ifindex) {
-  results.handler = DELETE_TUNNEL_INTERFACE;
-  results.num_handler_calls += 1;
-  results.ifindex = ifindex;
+  struct test_results temp = {0};
+  temp.handler = DELETE_TUNNEL_INTERFACE;
+  temp.ifindex = ifindex;
+  results.push_back(temp);
 }
 
 // fulfills an external reference
-int switchlink_create_vrf(switchlink_handle_t *vrf_h) {
+int switchlink_create_vrf(switchlink_handle_t* vrf_h) {
+  struct test_results temp = {0};
+  temp.handler = CREATE_VRF;
+  results.push_back(temp);
   return 0;
 }
 
-/**
- * Test fixture.
- */
+//----------------------------------------------------------------------
+// Test fixture
+//----------------------------------------------------------------------
+
 class SwitchlinkTest : public ::testing::Test {
  protected:
-  struct nl_msg *nlmsg_ = nullptr;
+  // Netlink message buffer.
+  struct nl_msg* nlmsg_ = nullptr;
 
   // Sets up the test fixture.
-  void SetUp() override {
-    ResetVariables();
-  }
+  void SetUp() override { ResetVariables(); }
```

Comment:
Also invoke `ResetVariables()` from `TearDown()` for cleanup

Reply:
This isn't a cleanup action, it's an initialization action.

I have renamed `ResetVariables()` to `InitializeGlobalState()` to make this fact abundantly clear.